hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/AlertsCriteria.java

```diff
@@ -122,28 +122,21 @@ public void setTriggerIds(Collection<String> triggerIds) {
         this.triggerIds = triggerIds;
     }
 
-    public Tag getTag() {
-        return tag;
-    }
-
-    /**
-     * @param tag fetched Alerts must be for triggers with the specified Tag. Ignored if Tags filter is set.
-     */
-    public void setTag(Tag tag) {
-        this.tag = tag;
-    }
-
-    public Collection<Tag> getTags() {
+    public Map<String, String> getTags() {
         return tags;
     }
 
-    /**
-     * @param tags fetched Alerts must be for trigger with any of the specified Tags.
-     */
-    public void setTags(Collection<Tag> tags) {
+    public void setTags(Map<String, String> tags) {
         this.tags = tags;
     }
 
+    public void addTag(String tname, String tvalue) {
```

Comment:
Minor comment: what about tagName, tagValue or simply name of value ? Just for consistency with the naming, in whole project we are using long variable names as tenantId os similar, I vote to maintain it. Or simply use name/value as with method is clear that we are talking about tags.